### PR TITLE
[Resubmit] Fix for incorrect usage of logging in torch/distributed/distributed_c10d.py

### LINKED ISSUE #51428: init_process_group changes the behavior of logging
When running init_process_group, Python library's logging is changed, as I show in the screenshot
Screenshot from 2021-01-31 10-17-57
This happens when I was using OpenSelfSup, a library made by CUHK, and the log disappeared. Because the framework use 'hasHandlers' function to judge whether the logging level is 'WARNING' or 'INFO', this behavior makes the logger level be 'WARNING' , all 'INFO' disappeared.
I don't know this is a bug or I did things wrong,
When I changed the source code version to 1.7.1 and recompile it again, this behavior was fixed.

# Enviroment
pytorch version: compiled from source, master branch
CUDA used to build Pytorch: 11.2
Python: 3.8.5
OS: ubuntu 2004
GPU: GeForce RTX 3090

Thanks so much for your attention.

cc @pietern @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @agolynski @SciPioneer @H-Huang @mrzzd @cbalioglu

### PR DESCRIPTION
**Pull request by szmigacz**
Resubmit of https://github.com/pytorch/pytorch/pull/51739
Fixes #51428

### CODE REVIEW

File: test/distributed/test_c10d.py [hunk 1]
```diff
@@ -578,6 +578,18 @@ def test_nominal(self):
...
         self.assertEqual(b"value0", store0.get("key0"))
 
+    @retry_on_connect_failures
+    def test_logging_init(self):
+        os.environ["WORLD_SIZE"] = "1"
+        os.environ["MASTER_ADDR"] = "127.0.0.1"
+        os.environ["MASTER_PORT"] = str(common.find_free_port())
+        os.environ["RANK"] = "0"
+
+        self.assertFalse(logging.root.hasHandlers())
```
Comment (pritamdamania87) [inline test/distributed/test_c10d.py:588]: This is not necessarily False in every environment. I think we should assert that the handlers before and after init_process_group are the same.

**Review by pritamdamania87 — CHANGES_REQUESTED** [review]
Requesting changes to fix unit test.

File: test/distributed/test_c10d.py (hunk 1 above)
Comment (szmigacz) [reply test/distributed/test_c10d.py]: @pritamdamania87 I modified the test to check individual handlers before and after the call to init_process_group

**Comment by szmigacz** [comment]
@pritamdamania87 I modified the unit test, please check now.

**Comment by pritamdamania87** [comment]
@szmigacz Looks like some of the CI jobs are failing with a merge conflict, could you rebase the PR?

**Comment by szmigacz** [comment]
@pritamdamania87 I rebased on top of latest master.

**Comment by pritamdamania87** [comment]
@szmigacz Looks like a couple of jobs are still failing due to merge conflicts, maybe you need another rebase?

```
See CircleCI build pytorch_linux_xenial_py3_6_gcc5_4_build (1/2)
Step: "(Optional) Merge target branch" (full log | diagnosis details | 🔁 rerun)

Automatic merge failed; fix conflicts and then commit the result.
See CircleCI build pytorch_xla_linux_bionic_py3_6_clang9_build (2/2)
Step: "(Optional) Merge target branch" (full log | diagnosis details | 🔁 rerun)

Automatic merge failed; fix conflicts and then commit the result.
```

**Comment by szmigacz** [comment]
I rebased again

**Comment by szmigacz** [comment]
@pritamdamania87 I rebased again.

**Comment by pritamdamania87** [comment]
@szmigacz Thanks! Will merge it in once all the CI jobs pass.

**Review by pritamdamania87 — APPROVED** [review]

**Pull request closed without merging**